csharp/src/Google.Protobuf/Reflection/Dynamic/DynamicMessage.cs

```diff
@@ -0,0 +1,377 @@
+﻿using Google.Protobuf.Collections;
+using System;
+using System.Collections;
+
+namespace Google.Protobuf.Reflection.Dynamic
+{
+
+    /// <summary>
+    /// An implementation of IMessage that can represent arbitrary types, given a MessageaDescriptor.
+    /// Unknown fields not supported.
+    /// </summary>
+    public sealed partial class DynamicMessage : IMessage
```

Comment:
Why is the class partial?

Reply:
Not required in our case, removed it. Other IMessage implementations were marked as partial, so added, but not required in our case.